### ISSUE TITLE
[Accessibility]: Code Insights — Screen reader doesn't announce dashboard creation state

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#41875]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33502

### Problem description

When you're creating a new dashboard screen reader doesn't announce a proper loading state while mutation for creation is still in progress (instead of it it says just "Loading") and right after dashboard was created it doesn't announce that it was created (instead it announce just dashboard name because you were redirected to the dashboard page with newly created dashboard.

### Expected behavior

It should
- Announce loading state properly (like creation new dashboard)
- Says something about finished state right after dashboard was created (we probably need to add something like toast UI on the dashboard page for it)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
